internal/atlassian/data_jira_issues.go

```diff
@@ -0,0 +1,170 @@
+package atlassian
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+
+	"github.com/blackstork-io/fabric/internal/atlassian/client"
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+	"github.com/blackstork-io/fabric/plugin/plugindata"
+)
+
+func makeJiraIssuesDataSource(loader ClientLoadFn) *plugin.DataSource {
+	return &plugin.DataSource{
+		Doc:      "Retrieve issues from Jira.",
+		DataFunc: searchJiraIssuesData(loader),
+		Config: &dataspec.RootSpec{
+			Attrs: []*dataspec.AttrSpec{
+				{
+					Name:        "domain",
+					Type:        cty.String,
+					Constraints: constraint.RequiredMeaningful,
+					Doc:         "Account Domain.",
+				},
+				{
+					Name:        "account_email",
+					Type:        cty.String,
+					Secret:      true,
+					Constraints: constraint.RequiredMeaningful,
+					Doc:         "Account Email.",
+				},
+				{
+					Name:        "api_token",
+					Type:        cty.String,
+					Secret:      true,
+					Constraints: constraint.RequiredMeaningful,
+					Doc:         "API Token.",
+				},
+			},
+		},
+		Args: &dataspec.RootSpec{
+			Attrs: []*dataspec.AttrSpec{
+				{
+					Name: "expand",
+					Type: cty.String,
+					Doc:  "Use expand to include additional information about issues in the response.",
+					OneOf: constraint.OneOf{
+						cty.StringVal("renderedFields"),
+						cty.StringVal("names"),
+						cty.StringVal("schema"),
+						cty.StringVal("changelog"),
+					},
+					ExampleVal: cty.StringVal("names"),
+				},
+				{
+					Name: "fields",
+					Type: cty.List(cty.String),
+					Doc:  "A list of fields to return for each issue.",
+					ExampleVal: cty.ListVal([]cty.Value{
+						cty.StringVal("*all"),
+					}),
+				},
+				{
+					Name:       "jql",
+					Type:       cty.String,
+					Doc:        "A JQL expression. For performance reasons, this field requires a bounded query. A bounded query is a query with a search restriction.",
+					ExampleVal: cty.StringVal("order by key desc"),
+				},
+				{
+					Name:         "properties",
+					Type:         cty.List(cty.String),
+					Doc:          "A list of up to 5 issue properties to include in the results.",
+					MaxInclusive: cty.NumberIntVal(5),
+				},
```

Comment:
If we add this here ...
```suggestion
					DefaultVal:   cty.ListValEmpty(cty.String),
				},
```